### DISCUSSION TITLE
Depend on cdpcli rather than cdpcli-beta

### DISCUSSION THREAD

**Discussion by paulmayer** [cloudera-labs/cdpy#96] [General]
Hi,

We are trying to make use of cdpy in some of our automation (indeed also ansible modules) and would ask whether:
- cdpy could be changed to depend on cdpcli rather than cdpcli-beta: installing cdpy in an environment where cdpcli is already present (at least with cdpcli==0.9.117 and cdpy==v1.3.0) causes conflicts in cdpcli/data (aliases.yaml may not include metrics: null, which causes the CLIDriver initialization to fail)
- cdpy could be published to pypi

pre-releases could of course still depend on cdpcli-beta rather than cdpcli, however, stable releases we believe would benefit to rely on cdpcli.

**Comment by wmudge** [comment]
Hello! Great points! Historically, we had to use the beta of `cdpcli` in order to get at some endpoints. I am not convinced anymore of this case, so moving to the GA version sounds reasonable.

This question raises a good point about dependencies, though. We have considered dropping the `cdpcli` dependency and working directly against the CDP Public Cloud endpoints directly, employing local functions to handle the request signing, similar to [cdpcurl](https://github.com/cloudera/cdpcurl). (See [CDP API Request Signing Specification V1](https://cloudera.github.io/cdp-dev-docs/api-docs/swagger/request_signing.html) for details.). 

As to publishing to Pypi, that would be ideal and something we have had on the back burner for some time. Stay tuned!

**Reply by paulmayer** [reply]
Hi, just wanted to inform you that there is a different [cdpy on pypi](https://pypi.org/project/cdpy) now, so the README instructions would now end up installing a different package for most users.

**Reply by wmudge** [reply]
Oh, yes, that should be updated. Thanks for the notice.
